src/pumipic_adjacency.tpp

```diff
@@ -544,5 +548,42 @@ namespace pumipic {
     Kokkos::Profiling::popRegion();
     return found;
   }
+
+  template <typename ParticleType, typename Segment3d>
+  struct NativeParticleHandlerAtElemBdry {
+    NativeParticleHandlerAtElemBdry () {}
```

Comment:
Based on above comments this functor will have some state data associated with it. I think the name can be something more descriptive about what this is actually doing. How about `RemoveParticleOnGeometricModelExit`?